File: Android/auikit-service/src/main/java/io/agora/auikit/model/AUIInvitationInfo.java
package io.agora.auikit.model;

import androidx.annotation.IntDef;
import androidx.annotation.NonNull;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

public class AUIInvitationInfo {

    private static final long kInvitationTimeoutTs = 10 * 1000;
    private static final long kInvitationInvalidTs = 20 * 1000;

    @IntDef({
            AUIInvitationType.Apply,
            AUIInvitationType.Invite,
    })
    @Retention(RetentionPolicy.RUNTIME)
    @Target(ElementType.FIELD)
    public @interface AUIInvitationType{
        int Apply = 1; // 观众申请
        int Invite = 2; // 主播邀请
    }

    @IntDef({
            AUIInvitationStatus.Waiting,
            AUIInvitationStatus.Accept,
            AUIInvitationStatus.Reject,
            AUIInvitationStatus.Timeout,
            AUIInvitationStatus.Cancel,
    })
    @Retention(RetentionPolicy.RUNTIME)
    @Target(ElementType.FIELD)
    public @interface AUIInvitationStatus{
        int Waiting = 1; // 等待确认
        int Accept = 2; // 同意
        int Reject = 3; // 拒绝
        int Timeout = 4; // 超时

        int Cancel = 5; // 取消
    }

    // 申请观众userId，被邀请观众userId
    public @NonNull String userId = "";

    // 麦位位置
    public int seatNo = 0;

    // 类型，申请 or 邀请
    public @AUIInvitationType int type = AUIInvitationType.Apply;

    public @AUIInvitationStatus int status = AUIInvitationStatus.Waiting;

    public long createTime = System.currentTimeMillis();

    public long editTime = System.currentTimeMillis();

    public long timeoutTs = kInvitationTimeoutTs;

    public long invalidTs = kInvitationInvalidTs;
}

File: iOS/AUIKitCore/Sources/Service/Model/AUIInvitationInfo.swift
//
//  AUIInvitationInfo.swift
//  AUIKitCore
//
//  Created by wushengtao on 2024/2/5.
//

import Foundation

let kInvitationTimeoutTs: Int64 = 10 * 1000
let kInvitationInvalidTs: Int64 = 20 * 1000

@objc public enum AUIInvitationType: Int {
    case apply = 1     // 观众申请
    case invite        // 主播邀请
}

@objc public  enum AUIInvitationStatus: Int {
    case waiting = 1   // 等待确认
    case accept = 2    // 同意
    case reject = 3    // 拒绝
    case timeout = 4   // 超时
    case cancel = 5    // 取消
}

@objcMembers public class AUIInvitationInfo: NSObject {
    public var userId: String = ""
    public var seatNo: Int = 0
    public var type: AUIInvitationType = .apply
    public var status: AUIInvitationStatus = .waiting
    public var createTime: Int64 = 0            //创建时间，和19700101的差，单位ms
    public var editTime: Int64 = 0              //编辑时间，和19700101的差，单位ms
    public var timeoutTs: Int64 = kInvitationTimeoutTs     //请求超时时间，单位ms
    public var invalidTs: Int64 = kInvitationInvalidTs    //无效数据(status非waitting)保存时间，单位ms
}
